infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -343,6 +376,65 @@ public void prefixTrim(long address) {
         logMetadata.prefixTrim(address);
     }
 
+    @Override
+    public void suffixTrim() {
+
+        // Have a global tail copy is necessary, since degrade to reset is dependent on it
+        long globalTailCopy = logMetadata.getGlobalTail();
+
+        try {
+            long committedTail = dataStore.getCommittedTail();
+
+            // Check if committed tail is not less than starting address
+            if (isTrimmed(committedTail)) {
+                throw new SuffixTrimException("local committed tail is trimmed",
+                        committedTail, globalTailCopy);
+            }
+
+            // Check if committed tail is greater than global tail
+            if (committedTail > globalTailCopy) {
+                throw new SuffixTrimException("local committed tail is greater than global tail",
+                        committedTail, globalTailCopy);
+            }
+
+            // Close all segment handlers and delete suffix log files
+            long committedSegment = committedTail / RECORDS_PER_LOG_FILE;
+            long tailSegment = Math.max(globalTailCopy, 0L) / RECORDS_PER_LOG_FILE;
+            closeSegmentHandlers(tailSegment);
+            deleteFilesMatchingFilter(file -> {
+                try {
+                    String segmentStr = file.getName().split("\\.")[0];
+                    long segment = Long.parseLong(segmentStr);
+                    return segment >= committedSegment && segment <= tailSegment;
+                } catch (Exception e) {
+                    log.warn("suffixTrim: ignoring file {}", file.getName());
+                    return false;
+                }
+            });
+
+            // Update tail segment
+            long newTailSegment = Math.max(committedSegment - 1, 0);
```

Comment:
I think it's a good idea to explain why the committed segment is also wiped out.